src/models/User.ts

```diff
@@ -0,0 +1,54 @@
+import mongoose, { Schema, Types, Document } from 'mongoose';
+import axios from 'axios';
+import bcrypt from 'bcrypt';
+
+export interface IUser extends Document {
+    firstname: string;
+    lastname: string;
+    username: string;
+    email: string;
+    password: string;
+    avatar?: string;
+}
+
+const userSchema = new Schema<IUser>({
+    firstname: { type: String, required: true },
+    lastname: { type: String, required: true },
+    username: { type: String, required: true },
+    email: { type: String, required: true },
+    password: { type: String, required: true },
+    avatar: { type: String, required: false },
+});
+
+userSchema.pre('save', async function (this: IUser, next: Function) {
+    const salt = await bcrypt.genSalt();
+    this.password = await bcrypt.hash(this.password, salt);
+    if (!this.avatar) {
+        const lockNumber = () => Math.floor(Math.random() * 99999) + 1;
+        const avatarUrl = await generateRandomUserAvatar(lockNumber());
+        this.avatar = avatarUrl;
+    }
+    next();
+});
+
+userSchema.methods.validatePassword = async function (passwordTry: string) {
```

Comment:
Can I call this function on User Objects?

Like `User1.validatePassword()`

Reply:
yes, this will add this method to all User objects

Reply:
ok thanks!

Reply:
I found here an example on how it's used https://github.com/oslabs-beta/Accumulus/blob/dev/server/database/db.ts#L27